src/snowflake/connector/connection.py

```diff
@@ -1541,13 +1543,27 @@ def is_an_error(status: QueryStatus) -> bool:
 
     def _all_async_queries_finished(self) -> bool:
         """Checks whether all async queries started by this Connection have finished executing."""
-        queries = copy.copy(
-            self._async_sfqids
-        )  # get_query_status might update _async_sfqids, let's copy the list
-        finished_async_queries = (
-            not self.is_still_running(self.get_query_status(q)) for q in queries
-        )
-        return all(finished_async_queries)
+
+        queries = list(reversed(list(self._async_sfqids.keys())))
```

Comment:
This could be simplified to `list(reversed(self._async_sfqids.keys()))` once we support 3.8+

Reply:
If so, how about?
```suggestion
        if sys.version_info >= (3, 8):
            queries = list(reversed(self._async_sfqids.keys()))
        else:
            queries = list(reversed(list(self._async_sfqids.keys())))
```
This makes refactoring this bit a child's play in the future.